### PR TITLE
Add new document on building filters, and accompanying utility functions

### PR DESCRIPTION
**Pull request by MikeTaylor**
Fixes STSMACOM-327.

### CODE REVIEW

File: lib/SearchAndSort/building-filters.md [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+* These values are used to provide the current selections in the `<MultiSelectionFilter>` ([line 199](https://github.com/openlibraryenvironment/ui-directory/blob/84f94d0a3414577fbb4903344d09ce4712d19d6e/src/routes/DirectoryEntries.js#L199)).
+
+
+### Stage 3.
+
+* [Line 200](https://github.com/openlibraryenvironment/ui-directory/blob/84f94d0a3414577fbb4903344d09ce4712d19d6e/src/routes/DirectoryEntries.js#L200) specifies that when a filter control changes, the `setFilterState` method should be called.
+* [line 194](https://github.com/openlibraryenvironment/ui-directory/blob/84f94d0a3414577fbb4903344d09ce4712d19d6e/src/routes/DirectoryEntries.js#L194) provides a callback for clearing the filter completely: the very simple `clearGroup` function provided in [line 183](https://github.com/openlibraryenvironment/ui-directory/blob/84f94d0a3414577fbb4903344d09ce4712d19d6e/src/routes/DirectoryEntries.js#L183), which merely invokes `setFilterState` with an empty set of valus.
```
Comment (EthanFreestone) [inline lib/SearchAndSort/building-filters.md:174]: Small typo, "valus" instead of "values"

Comment (MikeTaylor) [reply lib/SearchAndSort/building-filters.md]: Thanks, now fixed!

**Comment by MikeTaylor** [comment]
I interpret the silence of the nominated reviewers as tacit approval, and will proceed with the merge.

**Comment by MikeTaylor** [comment]
Thanks, @zburke, I really appeciate this! It's easy sometimes to feel that documentation updates fall through the cracks of people's (understandably stretched) attention — and really rewarding when a document does land right!

**Comment by mkuklis** [comment]
@MikeTaylor thank you for this!

**Pull request merged**